perun-core/src/main/java/cz/metacentrum/perun/core/impl/ExtSourceSql.java

```diff
@@ -358,6 +323,94 @@ public void close() {
 		}
 	}
 
+	/**
+	 * Decodes a full attribute name from the given value. For used mappings,
+	 * see attributeNameMapping.
+	 *
+	 * @param value which will be mapped
+	 * @return attribute full name
+	 */
+	private String mapAttributeNames(String value) {
+		String[] attributeRaw = value.split(":", 3);
+		String attributeName = null;
+		if (!attributeNameMapping.containsKey(attributeRaw[0])) {
+			log.warn("Unknown attribute type '{}', attributeRaw {}", attributeRaw[0], attributeRaw);
+		} else if (!attributeNameMapping.containsKey(attributeRaw[1])) {
+			log.warn("Unknown attribute type '{}', attributeRaw {}", attributeRaw[1], attributeRaw);
+		} else {
+			attributeName = attributeNameMapping.get(attributeRaw[0]) + attributeNameMapping.get(attributeRaw[1]) +
+					attributeRaw[2];
+		}
+		return attributeName;
+	}
+
+	/**
+	 * Add additional data from the given result set to the given map.
+	 *
+	 * @param map map with attributeName-value entries
+	 * @param rs result set with attribute data
+	 * @throws SQLException SQLException
+	 */
+	private void addAdditionalAttributeData(Map<String, String> map, ResultSet rs) throws SQLException {
```

Comment:
I would prefer to return the new resulting Map instead of just changing the input one.

Reply:
I have changed it so it returns the data.